src/components/calcite-picker/calcite-picker.tsx

```diff
@@ -0,0 +1,291 @@
+        const end = this.rows.indexOf(row);
+        this.rows.slice(Math.min(start, end), Math.max(start, end)).forEach((currentRow) => {
+          currentRow.setAttribute("selected", "");
+          this.selectedValues.add(currentRow);
+        });
+      } else {
+        this.selectedValues.add(row);
+      }
+    } else {
+      this.selectedValues.delete(row);
+    }
+    if (!this.multiple && selected) {
+      this.rows.forEach((item) => {
+        if (item !== row) {
+          this.deselectRow(item);
+        }
+      });
+    }
+    this.lastSelectedRow = row;
+    this.calcitePickerSelectionChange.emit(this.selectedValues);
+  }
+
+  @Listen("calcitePickerRowDeleted")
+  calcitePickerRowDeletedHandler(event): void {
+    event.stopPropagation(); // private event
+    const { row } = event.detail;
+    row.setAttribute("hidden", "");
+    this.deletedRows.add(row);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  setupRows(): void {
+    this.rows = Array.from(this.el.querySelectorAll("calcite-picker-row"));
+    this.rows.forEach((row) => {
+      row.setAttribute("icon", this.getIconType());
+    });
+    if (this.dragEnabled && this.mode === "configuration") {
+      this.setUpDragAndDrop();
+    }
+  }
+
+  setUpDragAndDrop(): void {
+    const sortGroups = [this.el, ...Array.from(this.el.querySelectorAll("calcite-picker-group"))];
+    sortGroups.forEach((sortGroup) => {
+      Sortable.create(sortGroup, {
+        group: this.el.id,
+        handle: `.${CSS.dragHandle}`,
+        draggable: "calcite-picker-row"
+      });
+    });
+  }
+
+  deselectRow(item: HTMLCalcitePickerRowElement): void {
+    item.removeAttribute("selected");
+    this.selectedValues.delete(item);
+  }
+
+  startEdit(): void {
+    this.editing = true;
+  }
+
+  cancelDelete(): void {
+    this.deletedRows.forEach((row: HTMLCalcitePickerRowElement) => {
+      row.removeAttribute("hidden");
+    });
+    this.deletedRows = new Set();
+    this.editing = false;
+  }
+
+  confirmDelete(): void {
+    let selectedChanged = false;
+    this.deletedRows.forEach((row: HTMLCalcitePickerRowElement) => {
+      if (this.selectedValues.has(row)) {
+        this.selectedValues.delete(row);
+        selectedChanged = true;
+      }
+      row.remove();
+    });
+    if (selectedChanged) {
+      this.calcitePickerSelectionChange.emit(this.selectedValues);
+    }
+    this.calcitePickerRowsDeleted.emit(this.deletedRows);
+    this.deletedRows = new Set();
+    this.editing = false;
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async getSelectedRows(): Promise<Set<object>> {
+    return this.selectedValues;
```

Comment:
I'm confused on why this method is needed if its only just returning `this.selectedValues`? Shouldn't `selectedValues` just be a property then?

Wouldn't it be more DOM-like for the user to just get the selected rows by doing this: `component.querySelectorAll("calcite-picker-item[selected=true]")`?

Reply:
It's generally accepted for components that properties shouldn't mutate. State mutates. This method exposes a way to get that state.

The data schema for querySelectorAll won't match what the event handler provides. This method will get back the same data exactly.

Reply:
Ok. @jcfranco is this ok? I'm just not seeing why we need this method. IMO we should go through dom queries to get the "data" which would just be dom elements.

Reply:
I think that the question is what is the source of truth? Is it the `row/item` nodes that have a selected attribute? In this code, it seems like the Picker is maintaining the state of what is selected, so there could be some kind of disconnect there. If we just relied on the row/item nodes being the source of truth for what is selected then this property wouldn't be necessary. right?

Reply:
I think this is another scenario where the end-user needs should drive our decision.
@asangma and @kerryrobinson can you do some research here and find out from Kevin, Adelheide and/or Kumar how they will be using this, and what information they are going to need from the picker? Will they be able to get by with querying dom nodes? Will they need an event listener to detect changes to selections in the picker? Will they need a method that can get them that data? Some questions to keep in mind is whether or not selection order is important - they won't be able to determine that if querying dom nodes, but the picker can track that state and surface it.

Reply:
I think putting this off until @asangma or @kat10140 follow up is good. API-wise, it does make sense to me to expose props/methods for this. I'd expect a property for this since it aligns with `HTMLSelectElement#selectedOptions` (read-only) and Stencil also recommends using props over methods.

Whether the item has a selected attribute or not for me it depends on the target API. Would we want something similar to how `HTMLSelectElement` + `HTMLOptionElement` work together or not?